pkg/resource/resource.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type KubernetesResource interface {
+	metav1.Object
+	runtime.Object
+}
```

Comment:
Why it should be added? What is the need/scenario that can be solved with?

Reply:
This can be an example where a generic method accepts any type of resource, sets the owner reference (metav1.Object) and creates the object (runtime.Object).

https://github.com/RHsyseng/operator-utils/blob/master/pkg/resource/write/writer.go#L62